python/TestHarness/schedulers/QueueManager.py

```diff
@@ -504,63 +305,267 @@ def augmentQueueParamsBase(self, job_container):
         template['queue_script'] = template['job_name'] + '.sh'
 
         # Call derived augmentQueueParams to fill in specifics
-        template = self.augmentQueueParams(job_container, template)
+        template = self.augmentQueueParams(job, template)
 
         return template
 
-    def waitFinish(self):
+    def reserveSlots(self, job):
         """
-        Inherited method called once all jobs have been sent to queueJobs.
-        QueueManager uses this method as a means to trigger other post-operation
-        methods after _all_ jobs have been launched. Such as calling the
-        queueJobs method a second time to perform status checks. As well as
-        writing the finalized json session file.
+        Inherited method which controls when jobs are allowed to execute,
+        depending on available resources.
+
+        QueueManager only executes third party queueing commands. So
+        modify every job to only require 1 process.
         """
+        job.setProcessors(1)
+        return Scheduler.reserveSlots(self, job)
 
-        # Set the status_check switch
-        if self.__status_check == False:
-            self.__status_check = True
+    def testOutput(self, job):
+        """
+        Adjust the Tester for a proper working directory (QueueManager creates sub-directores),
+        and allow derived Tester to perform processResults.
 
-            # TestHarness is finished sending us jobs, do postLaunch stuff (for PBS this
-            # means releasing the jobs using qrls)
-            self.postLaunch(self.__jobs)
+        Return resulting output.
+        """
+        tester = job.getTester()
+        json_job = self.getData(job.getUniqueIdentifier(), std_out=True, working_dir=True)
 
-            # Inform the user of our status change
-            print('\nJobs launched. Begin checking status...')
+        if json_job['std_out']:
+            stdout_file = os.path.join(json_job['working_dir'], json_job['std_out'])
+        else:
+            stdout_file=''
 
-            # Call queueJobs again using the populated set of jobs as our list (clever!
-            # no longer do we need to nerf the TestHarness's os.walk routine...)
-            self.queueJobs(status_jobs=self.__jobs)
+        if os.path.exists(stdout_file):
+            file = open(stdout_file, 'r+')
```

Comment:
Is it guaranteed that the block after this if/else never throws? You might want to use a `with` on the open
just in case (which would require a bit of code restructuring).